### DISCUSSION TITLE
FP16 slower than FP32

### DISCUSSION THREAD

**Discussion by soundarthiaga** [microsoft/onnxruntime#9282] [Show & Tell]
I noticed that FP16 model performs slower than FP32.
the issue is in graph optimization. 
The graph optimization is done first and then the typecasting is done. So, in the case of FP16 graph no optimization takes place.
But once we modify the code by typecasting first and then optimizing the FP16 graph gets optimized.

I have the code fix for this in inference_session.cc
Can you help me to push it into the  repo.

**Comment by ashbhandare** [comment]
Hi @soundarthiaga ,please make a PR so that the change can be understood/reviewed.
